common/cache/namespaceCache.go

```diff
@@ -492,16 +501,34 @@ UpdateLoop:
 	c.cacheByID.Store(newCacheByID)
 	c.cacheNameToID.Store(newCacheNameToID)
 	c.triggerNamespaceChangeCallbackLocked(prevEntries, nextEntries)
+
+	// only update last refresh time when refresh succeeded
+	c.lastRefreshTime.Store(now)
```

Comment:
Why `lastRefreshTime` is atomic? This func is called when lock is already occurred. I believe it is safe to use regular `time.Time` here.

Reply:
it is also used here:
https://github.com/temporalio/temporal/pull/824/files#diff-a1dd08cc18ae4897b834f91ace740ef18fcc17653d02e2b3ad59a9cfdf59f55aR515